src/python/pants/backend/java/lint/google_java_format/subsystem.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from typing import cast
+
+from pants.jvm.resolve.jvm_tool import JvmToolBase
+from pants.util.docutil import git_url
+
+
+class GoogleJavaFormatSubsystem(JvmToolBase):
+    options_scope = "google-java-format"
+    help = "Google Java Format (https://github.com/google/google-java-format)"
+
+    default_version = "1.13.0"
+    default_artifacts = ["com.google.googlejavaformat:google-java-format:{version}"]
+    default_lockfile_resource = (
+        "pants.backend.java.lint.google_java_format",
+        "google_java_format.default.lockfile.txt",
+    )
+    default_lockfile_path = "src/python/pants/backend/java/lint/google_java_format/google_java_format.default.lockfile.txt"
```

Comment:
This should be set only in our repo (via `pants.toml`, and a more natural default should be set for outside of it... probably something like `3rdparty/(jvm|tool)/${name}.lockfile`.

cc @Eric-Arellano, @chrisjrn : re: conventions

Reply:
Also `default_lockfile_path` isn't actually defined in `JvmToolBase` or `PythonToolBase`. This needs some thought about how lockfile paths and lockfile resource locations interact. It shouldn't be an issue in any event since the user would need to set the `--TOOL-lockfile` option to force use of a their own lockfile, at which point they would be specifying the path particular to their repo.

Reply:
Sure: but it is consumed as the default of the `JvmToolBase` option. The default could probably be computed via the tool name though.

Reply:
@tdyas : I don't think so, because `PythonToolBase` does not actually consume the `default_lockfile_path` field to generate the default value for `--lockfile` the way `JvmToolBase` does: https://github.com/pantsbuild/pants/blob/4c871edc37db2925f1d8647d5f16f272a923a4d3/src/python/pants/backend/python/subsystems/python_tool_base.py#L95-L120